samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -114,6 +100,11 @@ public boolean tryBecomeLeader() {
     String predecessor = children.get(index - 1);
     if (!predecessor.equals(currentSubscription)) {
       if (currentSubscription != null) {
+
+        // callback in case if the previous node gets deleted (when previous processor dies)
```

Comment:
What case are we handling here? The previous processor dying should have already trigger the PreviousProcessorChangeListener in the event thread. 

If the candidate you are subscribing to doesn't exist, we are handling it by double checking its existence in Line 122 and retying the leader election process.